### PR TITLE
Tests/time advance

### LINKED ISSUE #373: extend test coverage in time_advance-tests
currently, we test only that the wavelet space solution for each problem at a given timestep matches the MATLAB solution for that problem/timestep. we could fairly simply extend these tests to also test that the difference between wavelet space solution and analytic solution (for problems with analytic solution) also matches. this will ensure we have consistent analytic solutions generated at every timestep.

### PR DESCRIPTION
**Pull request by hbrunie**
## Proposed changes
Add tests in time advance in order to check that the simulated solution is close enough to the analytic one.
Fixes #373 .

## What type(s) of changes does this code introduce?

- [x] Bugfix

### Does this introduce a breaking change?

- [x] No

## What systems has this change been tested on?
Tested on my laptop (INTEL CPU on ubuntu).

## Checklist

- [x] this PR is up to date with current the current state of 'develop'
- [x] code added or changed in the PR has been clang-formatted
- [x] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [x] documentation has been added (if appropriate)

### CODE REVIEW

**Comment by hbrunie** [comment]
@quantumsteve I can't reproduce the CI bug locally. I tried to compile with -DASGARD_USE_MPI=ON but it did not change the tests compiled with make.
Do you have any insight on how to compile it? Thank you.

**Comment by quantumsteve** [comment]
```
$ ctest -R time_advance-test-mpi --verbose
UpdateCTestConfiguration  from :/home/svh/Documents/asgard/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/svh/Documents/asgard/build/DartConfiguration.tcl
Test project /home/svh/Documents/asgard/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 20
    Start 20: time_advance-test-mpi

20: Test command: /usr/bin/mpirun "-n" "4" "./time_advance-tests"
20: Test timeout computed to be: 10000000
20: Invalid MIT-MAGIC-COOKIE-1 key
20: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20: time_advance-tests is a Catch v2.11.3 host application.
20: Run with -? for options
20: 
20: -------------------------------------------------------------------------------
20: adaptive time advance - double
20:   diffusion 2 explicit
20: -------------------------------------------------------------------------------
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:266
20: 
20: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20: time_advance-tests is a Catch v2.11.3 host application.
20: Run with -? for options
20: 
20: -------------------------------------------------------------------------------
20: adaptive time advance - double
20:   diffusion 2 explicit
20: -------------------------------------------------------------------------------
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:266
20: ...............................................................................
20: 
20: /home/svh/Documents/asgard/testing/tests_general.hpp:35: FAILED:
20:   REQUIRE( (diff_norm / max) < (tolerance * std::sqrt(v0.size())) )
20: with expansion:
20:   4.862126855370838 < 4.698936049788292
20: 
20: 
20: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20: 
20: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20: time_advance-tests is a Catch v2.11.3 host application.
20: Run with -? for options
20: 
20: -------------------------------------------------------------------------------
20: adaptive time advance - double
20: ...............................................................................
20: 
20: /home/svh/Documents/asgard/testing/tests_general.hpp:35: FAILED:
20:   REQUIRE( (diff_norm / max) < (tolerance * std::sqrt(v0.size())) )
20: with expansion:
20:   6.671845945279083 < 4.715930449020639
20: 
20:   diffusion 2 explicit
20: -------------------------------------------------------------------------------
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:266
20: ...............................................................................
20: 
20: /home/svh/Documents/asgard/testing/tests_general.hpp:35: FAILED:
20:   REQUIRE( (diff_norm / max) < (tolerance * std::sqrt(v0.size())) )
20: with expansion:
20:   6.671845945279083 < 4.715930449020639
20: 
20: time_advance-tests is a Catch v2.11.3 host application.
20: Run with -? for options
20: 
20: -------------------------------------------------------------------------------
20: adaptive time advance - double
20:   diffusion 2 explicit
20: -------------------------------------------------------------------------------
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:266
20: ...............................................................................
20: 
20: /home/svh/Documents/asgard/testing/tests_general.hpp:35: FAILED:
20:   REQUIRE( (diff_norm / max) < (tolerance * std::sqrt(v0.size())) )
20: with expansion:
20:   4.862126855370838 < 4.698936049788292
20: 
20: -------------------------------------------------------------------------------
20: adaptive time advance - float
20:   diffusion 2 explicit
20: -------------------------------------------------------------------------------
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:266
20: ...............................................................................
20: 
20: /home/svh/Documents/asgard/testing/tests_general.hpp:35: FAILED:
20:   REQUIRE( (diff_norm / max) < (tolerance * std::sqrt(v0.size())) )
20: with expansion:
20:   5.099142074584961f < 4.418144472710386
20: 
20: -------------------------------------------------------------------------------
20: adaptive time advance - float
20:   diffusion 2 explicit
20: -------------------------------------------------------------------------------
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:266
20: ...............................................................................
20: 
20: /home/svh/Documents/asgard/testing/tests_general.hpp:35: FAILED:
20:   REQUIRE( (diff_norm / max) < (tolerance * std::sqrt(v0.size())) )
20: with expansion:
20:   5.099142074584961f < 4.418144472710386
20: 
20: [PC102795:48423] *** An error occurred in MPI_Allreduce
20: [PC102795:48423] *** reported by process [2123300865,3]
20: [PC102795:48423] *** on communicator MPI COMMUNICATOR 5 SPLIT FROM 0
20: [PC102795:48423] *** MPI_ERR_TRUNCATE: message truncated
20: [PC102795:48423] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
20: [PC102795:48423] ***    and potentially your MPI job)
20: -------------------------------------------------------------------------------
20: adaptive time advance - float
20:   continuity 2 explicit
20: -------------------------------------------------------------------------------
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:339
20: ...............................................................................
20: 
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:339: FAILED:
20: due to a fatal error condition:
20:   SIGTERM - Termination request signal
20: 
20: ===============================================================================
20: test cases:   4 |   2 passed | 2 failed
20: assertions: 164 | 161 passed | 3 failed
20: 
20: -------------------------------------------------------------------------------
20: adaptive time advance - float
20:   continuity 2 explicit
20: -------------------------------------------------------------------------------
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:339
20: ...............................................................................
20: 
20: /home/svh/Documents/asgard/src/time_advance_tests.cpp:339: FAILED:
20: due to a fatal error condition:
20:   SIGTERM - Termination request signal
20: 
20: ===============================================================================
20: test cases:   4 |   2 passed | 2 failed
20: assertions: 164 | 161 passed | 3 failed
20: 
20: [PC102795:48415] 1 more process has sent help message help-mpi-errors.txt / mpi_errors_are_fatal
20: [PC102795:48415] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
1/1 Test #20: time_advance-test-mpi ............***Failed   52.51 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =  52.51 sec

The following tests FAILED:
	 20 - time_advance-test-mpi (Failed)
Errors while running CTest
Output from these tests are in: /home/svh/Documents/asgard/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```

File: src/time_advance_tests.cpp [hunk 1]
```diff
@@ -211,6 +281,7 @@ TEST_CASE("adaptive time advance")
                        parser::DEFAULT_MAX_LEVEL, num_steps, use_implicit,
                        do_adapt_levels, adapt_threshold);
     time_advance_test(parse, gold_base, tol_factor);
+    time_advance_simulated_vs_analytic_test(parse, tol_rmse_factor);
```
Comment (quantumsteve) [inline src/time_advance_tests.cpp:284]: @hbrunie The issue is somewhere around here. All test pass if I comment out this line.

**Comment by mkstoyanov** [comment]
Stalled PR, feel free to resubmit based on the latest developments.

**Pull request closed without merging**